src/ui/AuthenticationProvider/AuthenticationProvider.ts

```diff
@@ -136,6 +144,56 @@ export class AuthenticationProvider extends Component {
     });
   }
 
+  private storeTemporaryTokenIfFoundInUrl() {
+    const token = this.getTemporaryTokenFromUrl();
+    token && localStorage.setItem(authProviderTemporaryToken, token);
+  }
+
+  private getTemporaryTokenFromUrl() {
+    const fragment = window.location.hash.slice(1);
+    const params = fragment.split('&');
+
+    const tokenParam = _.find(params, param => {
+      const [key] = param.split('=');
+      return key === 'access_token';
```

Comment:
is now `handshake_token`

we try to preserve the rest of hash when adding this but it'll be a good idea to test that out properly as well

Reply:
Adjusted the param name, and added a test case for when the token is part of a larger hash string,